src/models/stores.ts

```diff
@@ -0,0 +1,27 @@
+import { ProblemModel, ProblemModelType } from "./problem";
+import { UIModel, UIModelType } from "./ui";
+import { UserModel, UserModelType } from "./user";
+
+export interface IStores {
+  devMode: boolean;
+  problem: ProblemModelType;
+  user: UserModelType;
+  ui: UIModelType;
+}
+
+export interface ICreateStores {
+  devMode: boolean;
+  problem?: ProblemModelType;
+  user?: UserModelType;
+  ui?: UIModelType;
+}
+
+export function createStores(params?: ICreateStores): IStores {
+  return {
+    devMode: params && (params.devMode != null) ? params.devMode : false,
```

Comment:
I'm a little confused by the `(params.devMode != null)` code.  Is this to check if it exists in the params?  If it can be optional it should be marked optional in `ICreateStores`.